File: app/src/main/java/io/ipoli/android/reward/viewmodels/RewardViewModel.java
package io.ipoli.android.reward.viewmodels;

import io.ipoli.android.reward.data.Reward;

/**
 * Created by Polina Zhelyazkova <polina@ipoli.io>
 * on 5/31/16.
 */
public class RewardViewModel {

    private Reward reward;

    private boolean canBeBought;

    public RewardViewModel(Reward reward, boolean canBeBought) {
        this.reward = reward;
        this.canBeBought = canBeBought;
    }

    public Reward getReward() {
        return reward;
    }

    public boolean canBeBought() {
        return canBeBought;
    }
}

File: app/src/main/java/io/ipoli/android/reward/list/RewardViewModel.kt
package io.ipoli.android.reward.list

/**
 * Created by Venelin Valkov <venelin@ipoli.io>
 * on 8/2/17.
 */
data class RewardViewModel(val id: String,
                           val name: String,
                           val description: String,
                           val price: Int,
                           val canBePurchased: Boolean)